roll/__init__.py

```diff
@@ -143,7 +147,7 @@ def add_route(view):
             if protocol_class.ALLOWED_METHODS:
                 assert set(methods) <= set(protocol_class.ALLOWED_METHODS)
             self.routes.add(path, **payload)
-            self.loop.create_task(self.hook("route:add", path, view, **extras))
+            self.schedule(self.hook("route:add", path, view, **extras))
```

Comment:
We actually don't have the loop at hand when `route` is called, because the loop is only added when the app is started in a running env.

Other option is to have some "non async" hooks.